spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UdpServerListeningEvent.java

```diff
@@ -32,7 +32,7 @@
 
 	private final int port;
 
-	public UdpServerListeningEvent(UnicastReceivingChannelAdapter adapter, int port) {
+	public UdpServerListeningEvent(Object adapter, int port) {
```

Comment:
Gary, I don't understand this problem.
I don't see how `org.springframework.integration.ip.util` and `org.springframework.integration.ip.udp` packages use each other. Unless you have included to your report `test` directory as well...

Reply:
The event and the adapter reference each other (class tangle)

`DatagramPacketMessageMapper` uses `RegexUtils` and `TestingUtilities` referenced the `UnicastReceivingChannelAdapter` (package tangle)